packages/router/router.ts

```diff
@@ -1928,6 +1928,22 @@ export function createRouter(init: RouterInit): Router {
         }
         return;
       }
+    } else if (
+      init.basename !== undefined && init.basename !== "/" &&
+      isBrowser &&
+      typeof window?.location !== "undefined"
+    ) {
+      // Only the last trailing slash is replaced
+      const BASENAME_URL_REGEX = new RegExp("^" + init.basename.replace(/\/$/, '') + "([/|?.*|?.#]|$)", 'i');
+
+      if (!BASENAME_URL_REGEX.test(redirect.location)) {
```

Comment:
I think this should be lumped in with the above `if` statement, as I think this only applies to absolute URL redirects?